File: basics/roscpp/simple_pub/src/simple_pub.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int
main(int argc, char **argv)
{
	ros::init(argc, argv, "simple_pub_node");

	ros::NodeHandle nh;

	ros::Publisher pub = nh.advertise<std_msgs::String>("/talk", 1);

	//2Hz rate

	ros::Rate loop_rate(2);

	std_msgs::String msg;
	msg.data = "Hello World";

	while(ros::ok()){
		pub.publish(msg);
		ros::spinOnce();
		loop_rate.sleep();
	}

	exit(EXIT_SUCCESS);
}

File: basics/rospy/simple_pub_py/src/simple_pub.py
#! /usr/bin/env python

import rospy
from std_msgs.msg import String

def main():
    rospy.init_node("simple_pub_node")

    pub = rospy.Publisher("/talk", String, queue_size=1)

    msg = String()
    msg.data = "Hello World"

    # 2Hz Rate:

    rate = rospy.Rate(2)
    while(not rospy.is_shutdown()):
        pub.publish(msg)
        rate.sleep()

if __name__ == "__main__":

    try:
        main()
    except rospy.ROSInterruptException:
        pass
